pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -2963,6 +2961,25 @@ public String startReplaceSegments(String tableNameWithType, List<String> segmen
     return segmentLineageEntryId;
   }
 
+  // TODO: Add more conflict checks over segmentsTo later. For example, for APPEND table,
+  //       if the new segments from 2 batch jobs are overlapping, we reject one of job.
+  private static boolean isConflicted(List<String> segmentsFrom, LineageEntry lineageEntry, TableConfig tableConfig) {
+    if (!segmentsFrom.isEmpty()) {
+      // It's conflicted if there is any overlap between segmentsFrom.
+      return !Collections.disjoint(segmentsFrom, lineageEntry.getSegmentsFrom());
+    }
+    // For REFRESH table, it's conflicted if both segmentsFrom are empty.
+    if (isRefreshTable(tableConfig)) {
```

Comment:
I don't follow the logic here. Say I have a refresh table, and I scheduled multiple segment ingestion tasks in parallel, why should one task kill another?
IMO, we should only perform the disjoint check because having common source segments can result in duplicate data

Reply:
cc @snleee to help chime in, in case there are any corner cases for REFRESH table. Otherwise, I'll cool with just using disjoint()

Reply:
I can elaborate on the reason why multiple segment ingestion tasks cannot be done in parallel for a refresh table. That's mainly due to how the segment name is generated. 
For refresh table, since there is no time column specified, there will be no min/max value shown in the segment name; the generated segment name will be like `testTable_postfix_0`. The suffix `_0` only denotes the identifier for the segment within a batch ingestion job.
And since there is no relationship between the input raw file names and output segment names, there is no way to tell which raw file maps to which output segment within a batch. Thus, there is no way to backfill only a subset of segments for refresh table, i.e. if we want to update some data in one segment, we need to replace the whole data of it, because we don't know which segment we actually need to replace with the new data.
That's why if there are two ingestion tasks running in parallel for the same refresh table, either one of them should be banned (either the latter push job which detects the same segment names from `segmentFrom` field if forceCleanup = false, or the former push job with forceCleanup = true so that its lineage state will be marked as `reverted`).

Reply:
I think we should be okay to just use disjoint() method here since the purpose is just to find out whether there is some segment names appearing in multiple segment lineages or not. That doesn't harm to make the logic of the validation stricter.

Reply:
@jackjlli We should decouple the logic of lineage from table push type. Lineage is designed to achieve atomic segments replacement, and we don't want to mix it with push type which can make it very confusing. If we want to do atomic full table refresh, we should generate segments with different names, and put old segments as `segmentsFrom`. Empty `segmentsFrom` means I want to add new segments without replacing existing ones, and we should allow multiple of such operations in parallel.

Reply:
@Jackie-Jiang 

My reasoning was based on the following assumption:

```
For REFRESH table, each offline ingestion flow would push the full data snapshot.
```

If this is the case, the correct behavior should be blocking if 2 flows with `segmentsFrom=empty` are running at the same time because this can result in the REFRESH table having 2 snapshots of data. (Even if the input files are the same, we now put `timestamp` to the segment name so 2 flows will be uploaded)

If we can make the assumption that we have 1 flow per REFRESH table, this can only happen if some user runs 2 push flow at the same time for an empty table, which will not likely happen. Then, I see your point that it's cleaner if we don't handle special logic for different push types. I'm fine with removing special handling for REFRESH.

On the other hand, if you have REFRESH tables with multiple flows set up, I think that this model will be hard to support the consistent push feature because you will have the hard time differentiating the segments from one flow to another as @jackjlli mentioned above. IMO, we should not support such model (having 2+ flows pushing to a single REFRESH table).

Reply:
@jackjlli @snleee In order to handle such case, we should also check if `segmentsTo` are disjoint. We don't want to override the just pushed segments. These checks are already there for non-force-cleanup branch, not sure why it is not done in the same way for force-cleanup branch.

Reply:
sry folks, I was distracted from this PR. IIUC, we'd like to avoid the specific handling on refresh table and simply do the disjoint check on segmentsFrom for now. So I just left the UT cases in this PR. I'll probably close the PR if we still need more discussions to converge on this. but lemme know your thoughts.